addons/crm/models/crm_team.py

```diff
@@ -625,11 +627,45 @@ def _assign_and_convert_leads(self, force_quota=False):
             # Need to check that record still exists since the ids have been fetched at the begining of the process
             # Previous iteration has commited the change, records may have been deleted in the meanwhile
             leads_to_assign = self.env['crm.lead'].browse(leads_to_assign_ids).exists()
+
+            members_to_assign_with_preferred_domain = [m for m in members_to_assign if m.assignment_preferred_domain and literal_eval(m.assignment_preferred_domain)]
+            preferred_leads_per_member = {
+                member: leads_to_assign.filtered_domain(expression.AND([literal_eval(member.assignment_domain or '[]'), literal_eval(member.assignment_preferred_domain)]))
+                for member in members_to_assign_with_preferred_domain
+            }
```

Comment:
A bit of readability would be welcomed ^^ current line is hard to read
```suggestion
            preferred_leads_per_member = {
                member: leads_to_assign.filtered_domain(
                    expression.AND([
                        literal_eval(member.assignment_domain or '[]'),
                        literal_eval(member.assignment_preferred_domain)
                    ])
                ) for member in members_to_assign_wpref
            }
```